File: code_examples/4-Functions/6-By_Value_or_By_Reference/CPP/main.cpp
// Copyright (c) 2020 Mr. Coxall All rights reserved.
//
// Created by: Mr. Coxall
// Created on: Sep 2020
// This program uses a function by reference

#include <iostream>

void addOne(int &someNumber) {
    // this function adds one, by reference
    someNumber = someNumber + 1;
}

int main() {
    // this function uses a function by reference
    int someNumber;

    // input
    std::cout << "Enter a number: ";
    std::cin >> someNumber;

    // process
    addOne(someNumber);

    // output
    std::cout << "The number plus one is: " << someNumber << std::endl;

    std::cout << "\nDone." << std::endl;
}

File: code_examples/4-Functions/6-By_Value_or_By_Reference/CSharp/main.cs
/* Created by: Mr. Coxall
 * Created on: Sep 2020
 * This program uses a function by reference
*/

using System;

/*
 * The Program class
*/
class Program {
    static void AddOne(ref int someNumber) {
        // this function adds one to the parameter, by reference
        someNumber = someNumber + 1;
    }

    static void Main() {
        // this function uses a function by reference
        int someNumber;

        // input
        Console.Write("Enter a number: ");
        someNumber = Convert.ToInt32(Console.ReadLine());

        // process
        AddOne(ref someNumber);

        // output
        Console.WriteLine("The number plus one is: {0}", someNumber);

        Console.WriteLine("\nDone.");
    }
}
